front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx

```diff
@@ -15,62 +22,159 @@ import { Breadcrumb } from '@/ui/navigation/bread-crumb/components/Breadcrumb';
 import {
   useDeleteOneApiKeyMutation,
   useGetApiKeyQuery,
+  useInsertOneApiKeyMutation,
 } from '~/generated/graphql';
 
+const StyledInfo = styled.span`
+  color: ${({ theme }) => theme.font.color.light};
+  font-size: ${({ theme }) => theme.font.size.sm};
+  font-weight: ${({ theme }) => theme.font.weight.regular};
+`;
+
+const StyledInputContainer = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  gap: ${({ theme }) => theme.spacing(2)};
+  width: 100%;
+`;
+
 export const SettingsDevelopersApiKeyDetail = () => {
   const navigate = useNavigate();
   const { apiKeyId = '' } = useParams();
-  const [generatedApiKey] = useRecoilState(generatedApiKeyState);
-  const apiKeyQuery = useGetApiKeyQuery({
+
+  const setGeneratedApi = useGeneratedApiKeys();
+  const [generatedApiKey] = useRecoilState(
+    generatedApiKeyFamilyState(apiKeyId),
+  );
+
+  const [deleteApiKey] = useDeleteOneApiKeyMutation();
+  const [insertOneApiKey] = useInsertOneApiKeyMutation();
+  const apiKeyData = useGetApiKeyQuery({
     variables: {
       apiKeyId,
     },
-  });
-  const [deleteApiKey] = useDeleteOneApiKeyMutation();
-  const deleteIntegration = async () => {
-    await deleteApiKey({ variables: { apiKeyId } });
-    navigate('/settings/developers/api-keys');
+  }).data?.findManyApiKey[0];
+
+  const deleteIntegration = async (redirect: boolean = true) => {
```

Comment:
No need to specify the type as Typescript already knows with the `true` default value it will be a boolean:

```suggestion
  const deleteIntegration = async (redirect = true) => {
```